File: ruby/sum_square_difference.rb
def sum_of_squares(x, y)
  num = (x..y).to_a
  sum = 0
  num.each do |n|
    sum += (n ** 2)
  end
  sum
end

def square_of_sum(x, y)
  num = (x..y).to_a
  sum = 0
  num.each do |n|
    sum += n
  end
  sum ** 2
end

x = 1
y = 100
sosq = sum_of_squares(x, y)
sosu = square_of_sum(x, y)
sum_square_difference = sosu - sosq

puts "#{sosu} - #{sosq} = #{sum_square_difference}"

File: javascript/sumSquareDifference.js
let sumSquare = (u) => {
  let sum = 0;
  for (i = 0; i <= u; i++) {
    sum += i**2;
  }
  return sum;
}

let squareSum = (v) => {
  let sum = 0;
  for (i = 0; i <= v; i++) {
    sum += i;
  }
  return sum**2;
}

let sumSquareDifference = (w) => {
  let sqd = squareSum(w) - sumSquare(w);
  return sqd;
}

let num = 100;
let x = sumSquare(num);
let y = squareSum(num);
let z = sumSquareDifference(num);

console.log(x);
console.log(y);
console.log(z);
